storagemarket/impl/provider.go

```diff
@@ -262,8 +262,8 @@ func (p *Provider) ListLocalDeals() ([]storagemarket.MinerDeal, error) {
 	return out, nil
 }
 
-func (p *Provider) AddAsk(price abi.TokenAmount, duration abi.ChainEpoch) error {
-	return p.storedAsk.AddAsk(price, duration)
+func (p *Provider) AddAsk(price abi.TokenAmount, duration abi.ChainEpoch, bounds ...abi.PaddedPieceSize) error {
```

Comment:
So I actually had something slightly different in mind when I mentioned variadic arguments -- I was talking about the functional options pattern:
https://www.sohamkamani.com/golang/options-pattern/

I would define the function this way:

```golang
type StorageAskOption func(*storagemarket.StorageAsk)
func (p *Provider) AddAsk(price abi.TokenAmount, duration abi.ChainEpoch, options ...StorageAskOption)
```

and then you can define:

```golang
func MinPieceSize(minPieceSize abi.PaddedPieceSize) StorageAskOptions {
   return func(sa *storagemarket.StorageAsk) {
      sa.MinPieceSize = minPieceSize
   }
}
```

And then, in storedAsk.AddAsk, you can just do:

```golang
ask := &storagemarket.StorageAsk{
		Price:        price,
		Timestamp:    height,
		Expiry:       height + duration,
		Miner:        s.actor,
		SeqNo:        seqno,
		MinPieceSize: defaultMinPieceSize,
                  MaxPieceSize: defaultMaxPieceSize,
}
for _, option := range options {
   option(ask)
}
```